### ISSUE TITLE
backend not launched in the electron app

### ISSUE THREAD

**Issue by kswang1029** [CARTAvis/carta-backend#533]
Over the past weeks (probably starting from v1.3) I noticed that _sometimes_ the backend is not launched successfully when we launch the electron app. The backend log is empty so we don't know what went wrong. It is hard to find a robust way to reproduce this... 

<img width="1392" alt="Screen Shot 2020-06-02 at 10 09 06 AM" src="https://user-images.githubusercontent.com/20819712/83472319-8bd33080-a4b9-11ea-908d-30d6c253d907.png">

(may consider to mark this issue as a critical bug)

**Comment by kswang1029** [comment]
It happened again and through the system monitor I do see carta_backend is running. So maybe it somehow lost websocket connection initially?
<img width="1277" alt="Screen Shot 2020-06-11 at 11 08 43 AM" src="https://user-images.githubusercontent.com/20819712/84340910-64c0e100-abd4-11ea-9c3d-4631dbda97df.png">


@ajm-asiaa could it be due to the electron wrapper?

**Comment by ajm-ska** [comment]
@kswang1029 My first guess is maybe port 55131 is already in use for some reason? 
It easy to check with the lsof command: `lsof -i :55131`

**Comment by kswang1029** [comment]
I don't see any output for now and CARTA can be launched successfully. Next time I will give a try.

**Comment by kswang1029** [comment]
I confirm it is not due to port. I tried to check the port usage before I tried to launch the electron app. 

/Applications/CARTA.app/Contents/Resources/app/carta-backend/bin/carta_backend: Version 1.3
Listening on port 55131 with root folder /, base folder /Users/kswang, 4 threads in worker thread pool and 16 OMP threads
Session 1 (Fri Jun 12 10:38:44 2020): Client 1 [::1] Connected. Num sessions: 1
Session 1 (Fri Jun 12 10:38:51 2020): Client 1 [::1] Disconnected. Remaining sessions: 1
No remaining sessions.

10:38:51 is the time I closed the electron window.

**Comment by ajm-ska** [comment]
That's strange. Does the light stay orange, or does it eventually go red?

**Comment by kswang1029** [comment]
no, it stays as orange, suggesting the initial connection was not successful but later trial was successful. Could it be due to somehow the backend took slightly longer time to be fully initialized and before that the frontend was trying to connect it?

**Comment by ajm-ska** [comment]
That's what I'm thinking. It seems it is getting stuck somehow. Isn't there a resume session dialog box that appears when it reconnects to the backend? Maybe that box is not appearing because the rest of the frontend is not loading up? I bet @veggiesaurus might know.

**Comment by kswang1029** [comment]
How are frontend and backend initialized? Asynchronously?

**Comment by veggiesaurus** [comment]
> That's what I'm thinking. It seems it is getting stuck somehow. Isn't there a resume session dialog box that appears when it reconnects to the backend? Maybe that box is not appearing because the rest of the frontend is not loading up? I bet @veggiesaurus might know.

I think what's happening is the frontend thinks it's _reconnecting_, rather than connecting. One solution I can think of is to delay the frontend loading until the backend is ready. A more useful solution would be to handle the _first_ successful connection as as a connection, and not a re-connection.

**Comment by veggiesaurus** [comment]
> no, it stays as orange, suggesting the initial connection was not successful but later trial was successful. Could it be due to somehow the backend took slightly longer time to be fully initialized and before that the frontend was trying to connect it?

@kswang1029 I think the light is orange until the first connection takes place. So perhaps it really hasn't connected? what does the tooltip say?

**Comment by kswang1029** [comment]
Luckily I can reproduce it just now.
<img width="433" alt="Screen Shot 2020-06-12 at 4 47 43 PM" src="https://user-images.githubusercontent.com/20819712/84483846-800f1780-accc-11ea-9ead-8a58798de901.png">

**Comment by veggiesaurus** [comment]
ok, I'll try fix that then. I think it's more of a frontend issue then

**Comment by kswang1029** [comment]
Weirdly it only happens when my computer went sleep long enough. Might have something to do with cache?
